sql_request_abstract/models/sql_request_mixin.py

```diff
@@ -0,0 +1,255 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2015 Akretion (<http://www.akretion.com>)
+# Copyright (C) 2017 - Today: GRAP (http://www.grap.coop)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import re
+import uuid
+import StringIO
+import base64
+from psycopg2 import ProgrammingError
+
+from openerp import _, api, fields, models
+from openerp.exceptions import Warning as UserError
+
+
+class SQLRequestMixin(models.Model):
+    _name = 'sql.request.mixin'
+
+    _clean_query_enabled = True
+
+    _check_prohibited_words_enabled = True
+
+    _check_execution_enabled = True
+
+    _sql_request_groups_relation = False
+
+    _sql_request_users_relation = False
+
+    STATE_SELECTION = [
+        ('draft', 'Draft'),
+        ('sql_valid', 'SQL Valid'),
+    ]
+
+    PROHIBITED_WORDS = [
+        'delete',
+        'drop',
+        'insert',
+        'alter',
+        'truncate',
+        'execute',
+        'create',
+        'update',
+        'ir_config_parameter',
+    ]
+
+    # Default Section
+    @api.model
+    def _default_group_ids(self):
+        ir_model_obj = self.env['ir.model.data']
+        return [ir_model_obj.xmlid_to_res_id(
+            'sql_request_abstract.group_sql_request_user')]
+
+    @api.model
+    def _default_user_ids(self):
+        return []
+
+    # Columns Section
+    name = fields.Char('Name', required=True)
+
+    query = fields.Text(
+        string='Query', required=True, help="You can't use the following words"
+        ": DELETE, DROP, CREATE, INSERT, ALTER, TRUNCATE, EXECUTE, UPDATE")
```

Comment:
all changes done, except this remark, because the function works correctly if it ends with or without ";" char. (the sql request will be cleaned)